crawl-ref/source/dat/des/altar/ecumenical.des

```diff
@@ -890,3 +890,149 @@ xxx....vlv....xxx
+MAP
+xxxxxxxxxxxxxxx
+xxxxxxt_txxxxxx
+xxxxp.!...pxxxx
+xxxp..?W?..pxxx
+xxxx..?W?..xxxx
+xxxxxp...pxxxxx
+xxxxxxx.xxxxxxx
+xxxxxp.xxpxxxxx
+xxxxxxp...xxxxx
+xx..pxxxxp.xxxx
+x.xx....xxx.pxx
+xp.xxxxx...pxxx
+xx.x...xxxxxxxx
+xxx.xpx.x..xxxx
+xxxxxxxp.xx.pxx
+xxxxxxxxxxxx@xx
+ENDMAP
+
+NAME:  dreamdust_ecumenical_altar_angels_and_demons
+TAGS:  transparent
+KFEAT: p = granite_statue
+KFEAT: q = granite_statue
+TILE:  G = dngn_crumbled_column
+TILE:  p = dngn_statue_angel
+TILE:  q = dngn_statue_demonic_bust
+FTILE: -_Gpq = floor_hall
+: ecumenical_altar_setup(_G)
+MAP
+.......................
+.........ccccc.........
+....ccccccp-pcccccc....
+.pcccG-G-G---G-G-Gcccq.
+@.+--------_--------+.@
+.pcccG-G-G---G-G-Gcccq.
+....ccccccq-qcccccc....
+.........ccccc.........
+.......................
+ENDMAP
+
+NAME:  dreamdust_ecumenical_altar_drowned_temple
+TAGS:  transparent
+MONS:  spectral goblin / spectral hobgoblin / \
+       spectral kobold / spectral gnoll 
+TILE:  G : dngn_statue_angel / dngn_statue_demonic_bust / \
+           dngn_statue_wraith / dngn_statue_mermaid / \
+           dngn_statue_ancient_hero / dngn_statue_ancient_evil
+SUBST: . = . W
+SUBST: - = - W
+SUBST: ~ = W:20 .
+SUBST: | = p:20 W
+SUBST: x = x:35 W
+FTILE: -1 = floor_pebble
+KFEAT: p = granite_statue
+TILE:  p = dngn_crumbled_column
+: ecumenical_altar_setup(_G)
+MAP
+.................
+..xxxx.....xxxx..
+..xG~xxxxxxx~Gx..
+..x~~~~~_~~~~~x..
+..xxx~|~1~|~xxx..
+ ...x~~~-~~~x...
+  ..x~|~-~|~x..
+  ..x~~~-~~~x..
+  ..x~|~-~|~x..
+  ..x~~~-~~~x..
+  ..x~|~-~|~x..
+  ..xx~~-~~xx..
+   ..xx+++xx..
+    .........
+     ..@@@..
+ENDMAP
+
+NAME:  dreamdust_ecumenical_altar_grove
+TAGS:  transparent
+MONS:  butterfly
+KMONS: p = plant / fungus / toadstool
+SUBST: # = g t p 1:5
+SUBST: - = .:20 g t p 
+SUBST: . = .g
+FTILE: _gtp1 = floor_grass
+: ecumenical_altar_setup(_G)
+MAP
+ ....... 
+.-------.
+.-#####-.
+.-#####-.
+.-##_##-.
+.-#####-.
+.-#####-.
+.-------.
+ ....... 
+ENDMAP
+
+NAME:   dreamdust_ecumenical_altar_sealed_demon
+TAGS:   no_monster_gen no_item_gen no_rotate
+WEIGHT: 5
+KMONS:  ? = any demon perm_ench:petrified
+KMONS:  p = withered plant
+TILE:   G = dngn_crumbled_column
+: set_feature_name("granite_statue", "a crumbling pillar")
+FTILE:  G_vn?- = floor_demonic_red
+TILE:   c = wall_stone_white
+TILE:   v = dngn_silver_wall
+: ecumenical_altar_setup(_G)
+MAP
+.........
+...ccc...
+.ccc_ccc.
+.cG---Gc.
+.c-vnv-c.
+.c-n?n-c.
+.c-vnv-c.
+.cG---Gc.
+.ccc+ccc.
+...p.p...
+....@....
+ENDMAP
+
+NAME:  dreamdust_ecumenical_altar_statues
+TAGS:  no_rotate transparent
+TILE:  G : DNGN_STATUE_CENTAUR / DNGN_STATUE_SNAIL / DNGN_STATUE_CAT / \
+           DNGN_STATUE_MERMAID / DNGN_STATUE_NAGA / DNGN_STATUE_DRAGON / \
+           DNGN_STATUE_ANGEL / DNGN_STATUE_ELEPHANT / DNGN_STATUE_WRAITH / \
+           DNGN_STATUE_IMP / DNGN_STATUE_CEREBOV / DNGN_STATUE_DEMONIC_BUST / \
+           DNGN_STATUE_ORB_GUARDIAN / DNGN_STATUE_AXE / DNGN_STATUE_POLEARM / \
```

Comment:
It is probably possible to match up text and tile customization here with lua but it will be a bit involved. If you have time and the inclination have at it, otherwise I will take a shot at it when merging.